tests/tests_geomstats/test_product_manifold.py

```diff
@@ -9,277 +11,286 @@
-
-    @geomstats.tests.np_and_autograd_only
-    def test_exp_log_vector(self):
-        n_samples = 5
-        expected = self.space_vector.random_point(n_samples)
-        base_point = self.space_vector.random_point(n_samples)
-        logs = self.space_vector.metric.log(expected, base_point)
-        result = self.space_vector.metric.exp(logs, base_point)
-        self.assertAllClose(result, expected)
-
-    @geomstats.tests.np_autograd_and_torch_only
-    def test_exp_log_matrix(self):
-        n_samples = 5
-        expected = self.space_matrix.random_point(n_samples)
-        base_point = self.space_matrix.random_point(n_samples)
-        logs = self.space_matrix.metric.log(expected, base_point)
-        result = self.space_matrix.metric.exp(logs, base_point)
+from tests.conftest import Parametrizer, TestCase, TestData, tf_backend
+
+space_matrix = ProductManifold(
+    manifolds=[Hypersphere(dim=2), Hyperboloid(dim=2)],
+    default_point_type="matrix",
+)
+space_vector = ProductManifold(
+    manifolds=[Hypersphere(dim=2), Hyperboloid(dim=3)], default_point_type="vector"
+)
+
+manifolds = [space_matrix, space_vector]
+
+SO3_2 = NFoldManifold(SpecialOrthogonal(3), 2)
+
+n_samples = 4
+SO3_2_point = gs.stack([gs.eye(3)] * SO3_2.n_copies * n_samples)
+SO3_2_point = gs.reshape(SO3_2_point, (n_samples, *SO3_2.shape))
+SO3_2_tangent_vec = SO3_2.to_tangent(gs.zeros((n_samples, *SO3_2.shape)), SO3_2_point)
+
+
+class TestProductManifold(TestCase, metaclass=Parametrizer):
+    cls = ProductManifold
+
+    class TestDataProductManifold(TestData):
+        def dimension_data(self):
+
+            smoke_data = [
+                dict(
+                    manifold=space_matrix,
+                    expected=4,
+                ),
+                dict(
+                    manifold=space_vector,
+                    expected=5,
+                ),
+            ]
+            return self.generate_tests(smoke_data)
+
+        def random_and_belongs_data(self):
+            n_samples = [1, 5, 10]
+            smoke_data = [
+                dict(manifold=manifold, n_samples=n_samples)
+                for (manifold, n_samples) in itertools.product(manifolds, n_samples)
+            ]
+            return self.generate_tests(smoke_data)
+
+        def exp_log_data(self):
+            n_samples = [5, 10]
+            smoke_data = [
+                dict(manifold=manifold, n_samples=n_samples)
+                for (manifold, n_samples) in itertools.product(manifolds, n_samples)
+            ]
+            return self.generate_tests(smoke_data)
+
+        def dist_log_exp_norm_data(self):
+            n_samples = [1, 5, 10]
+            smoke_data = [
+                dict(manifold=manifold, n_samples=n_samples)
+                for (manifold, n_samples) in itertools.product(
+                    [manifolds[0]], n_samples
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def inner_product_data(self):
```

Comment:
This should be `inner_product_matrix_data`: we are checking that the inner product matrix is correct, e.g. it is gs.eye(6).